src/core/tsi/ssl_transport_security.cc

```diff
@@ -377,7 +374,7 @@ static tsi_result add_subject_alt_names_properties_to_peer(
       result = tsi_construct_string_peer_property(
           TSI_X509_SUBJECT_ALTERNATIVE_NAME_PEER_PROPERTY,
           reinterpret_cast<const char*>(name), static_cast<size_t>(name_size),
-          &peer->properties[peer->property_count++]);
+          &peer->properties[(*current_insert_index)++]);
```

Comment:
why we need line 402-419 that extracts URI string again? 
why not just add here (to avoid extract name and name_size again)

```
if (subject_alt_name->type == GEN_RUI) {
      result = tsi_construct_string_peer_property(
          TSI_X509_URI_PEER_PROPERTY, reinterpret_cast<const char*>(name),
          static_cast<size_t>(name_size),
          &peer->properties[(*current_insert_index)++]);
}
```

Reply:
Yeah that's another option. I choose the other way because we have a `if (result != TSI_OK) break;` check at the end of this big if-else block. If we pile the two `tsi_construct_string_peer_property` at the same block, we might want to do an additional tsi_result check in between, and free the memory if necessary.
I don't have a strong preference of the two ways, and the newly pushed commit is the new version I described above. Let me know if it works:)